### ISSUE TITLE
Double Duration Bug

### ISSUE THREAD

**Issue by peyre** [raceintospace/raceintospace#246]
When I played today, I saw something strange in Future Missions.  Setting a duration on a mission seems to assign a 10-point Safety penalty!  I hadn't flown a suborbital yet, so 4 of those points were expected, but the other 10?

![Super Duration Penalty](https://user-images.githubusercontent.com/3118014/64492500-2d7a6980-d229-11e9-8dbf-03561313a6d0.png)

**Comment by pemensik** [comment]
I think that would be correct. Duration is next goal after you have been on orbit. It should be far difficult and I think it requires more advanced craft.

**Comment by peyre** [comment]
No, that can't be right.  The game never did this before.  The way it's supposed to work is that if you're penalized 5 points if you skip a Duration level.  So, if you've flown a Manned Orbital (which is Dur A), then schedule a Duration C mission, you'd be penalized 5 points for skipping B.  There shouldn't be any duration penalty assigning your first Duration B mission (or, at most, you should get a 5-point penalty if you haven't flown any missions at all).

There used to be what we called the Duration B Bug.  It goes all the way back to the original floppy version 1.00 of BARIS.  The game would _always_ penalize you 5 points for your first Duration mission, no matter what (your first Duration mission was usually level B, hence the name).

**Comment by peyre** [comment]
Hm, I thought maybe I was experiencing a fluke, that it might be happening because I was using an old save with a current version of the software, but no, I just started fresh and am experiencing the same issue.

So, the situation is: I haven't flown a suborbital yet, and when I plan the orbital for next turn, I show a -4 (as I should), but if I schedule a Duration B, I show a -14.  That I shouldn't be seeing.  It would be ok to show a -9 at this point, which would be:
-1 for it being a mission type I haven't flown before
-3 for skipping a step (since I haven't flown a suborbital yet)
-5 for skipping Duration A and going straight from no flight time at all directly to Dur B

The idea with Duration penalties is you're not supposed to skip, say, from having flown a Dur A to flying a Dur C without flying a Dur B in between.  I don't think that was supposed to apply when you schedule a Dur B without having flown an A yet, but I would be ok with the game doing that if we want it to.  But it shouldn't apply _two_ penalties when you're skipping _one_ duration level.

We disabled the Duration penalty system a while back because it was malfunctioning, and it seems to be doing so again.  I want to see it either fixed or disabled.  The way it malfunctions is the worst of both worlds.

**Comment by peyre** [comment]
So, here's some more info.  I experienced the Duration Penalty issue later in the game when I scheduled a Manned Lunar Pass (Dur C) without having flown Duration B, I think.  Before I hadn't been certain the penalty would actually be applied during the mission, but as you can see the rolls during the mission did reflect the penalty.

![Manned Lunar Pass](https://user-images.githubusercontent.com/3118014/64911233-e7635100-d6d3-11e9-9d76-64c24d79ae2f.png)
![Duration Penalty](https://user-images.githubusercontent.com/3118014/64911238-ed593200-d6d3-11e9-9638-3aee6acb88e7.png)

**Comment by rnyoakum** [comment]
Okay, I can explain this.

The code for calculating the mission penalty is handled by the PrestMin function in prest.cpp. There are a few issues with its methodology that we might want to address. But here's what is going on:

I) Every mission starts at a baseline -0 penalty. Yay! We're going straight to the moon!
II) It calculates the highest level milestone the mission requires. In your example, Milestone_EarthOrbit. The game then adds a -3 penalty for each milestone beneath that which wasn't achieved. Eh, maybe a lunar landing was pushing it.
III) The total milestone penalty is reduced for lower-difficulty players (2 points easier for Lvl I, 1 point for Lvl II) to a minimum total penalty of -0.
IV) If the mission is not a Manned suborbital, Manned orbital, or Manned orbital EVA, it applies a duration penalty if the duration requirement is **more than** one duration level beyond current achievement. That penalty is 5 x the duration deficit.

What's troubling you is step IV. You don't see a Duration A penalty because Duration A is the next duration landmark, but once you look beyond that at Duration B the game adds back in the Duration A penalty. Likewise, if you had achieved Duration B, you wouldn't see a penalty for Duration C, but the C penalty would be applied if you were going for Duration D.

This is an easy issue to fix. Leon, how would you like Duration penalties to be handled? To me the most logical choices would be

I) No duration penalty for the next duration step & a -5 penalty for each step after that, or
II) No duration penalty for duration A, but a -5 for every unmet duration goal after that (which would always add a -5 for your first Duration B, but not hinder Duration A missions)

but if there's a different logic you prefer, that can be added.

Now it's time for the methodology concerns I mentioned earlier. If you agree with me, I can create new issues for them:
I) The fifth milestone is a Lunar Probe Landing, the sixth a Lunar Pass, and the seventh a Lunar Orbit. This means your Lunar Pass & Lunar Orbit suffer a -3 penalty if you haven't landed a Lunar Probe. Which is ... weird. The tricky part is that Lunar Probe & Lunar Pass/Orbit are logically separate branches that come together for a Lunar Landing but the game's milestone system is linear. It's possible to disentangle them, but would probably require coding some special cases. More concerning is that I don't know if, or how, it would affect the AI.
II) In the Future Missions screen, when I change the duration setting on a duration mission, the penalty display does not update.
III) Currently, player difficulty adjustments only offset Milestone penalties. Ex: if you've gotten the Earth Orbital milestone and are doing an Orbital Duration mission (which would have duration penalty but no new milestone penalty), you don't get a Player Difficulty adjustment (step III) on the duration penalty. Should you?

**Comment by peyre** [comment]
Very thorough, Ryan.  Your explanation for how the penalty system works matches my understanding of it.  Re. getting a duration penalty if you go straight to B without having flown a mission: I'm not certain if it's supposed to work that way, since the duration penalty system has never worked properly in the game.  I think it's probably proper that basically skipping Duration A, flying a Dur B mission when you haven't had a spaceflight yet, would get the penalty.  If I've been unclear, the problem I have with this issue is that the game seems to be applying **two** duration penalties when I'm only skipping one duration level.

Yes, as you mention, the milestone penalty system is linear, which of course is a holdover from _Liftoff!_.  To clarify, if you haven't flown a lunar probe landing, your Lunar Pass will suffer a penalty, but if you fly that successfully, then the Lunar Orbital won't suffer a penalty, since accomplishing a milestone gives you credit for all previous milestones.

I think it's worth considering making the Lunar Probe Landing a milestone specifically for the Lunar Landing.  Logically, a probe landing wouldn't matter if you're not landing, and if we did this it would give the player more incentive to do a probe landing.  I could be persuaded otherwise, but that's my current thinking.

Future Missions isn't updating the duration penalty?  If I remember right, you have to change the mission type in order for the penalty to change (say, go up one and then down one).  If you're doing that and it's not updating, you might have found another bug).

My understanding is that player difficulty adds one or two -3 penalties to each mission, as if you were suffering permanent milestone penalties.  It shouldn't affect duration penalties.

**Comment by rnyoakum** [comment]
I understand both what the game is doing and why it's doing it. What I'm not certain about is what it should be doing.

Per the duration penalty - I described, perhaps confusingly, precisely how the system currently works. It subtracts the highest achieved duration from the mission duration and multiplies that value by five. However, the Manned suborbital, Manned orbital, and Manned orbital EVA missions are specifically exempted and suffer no duration penalty (helping ensure Yuri can get to space). Thus, a Manned orbital may be completed without any duration penalty to achieve an A duration, but attempting a different mission with an A duration first would incur a -5 penalty.

In the original code, each step of duration beyond the highest achieved duration adds -5 to the penalty. Thus, after earning an A duration (e.g. via a Manned orbital), attempting a B duration would incur a -5 penalty and a C duration a -10 penalty.

At some point a change was made to ignore that first level of penalty. Consequently, attempting a B duration would have no additional penalty for a highest achieved duration of A, an achieved B would have no penalty for C, and so forth. More specifically, it ignores the duration penalty for any mission of one higher level than achieved. However, that modification did not change how the penalty was calculated; it simply did not apply it. Thus, using the same example, a C duration still has a -10 penalty, but a B duration would have no penalty applied.

Based on this, the question becomes what the logic _should_ be.
I) Each level of duration beyond the highest achieved imposes a -5 penalty. In this scenario, achieving a B duration would allow any B duration mission to be accomplished with no penalty, a C mission with -5, a D mission with -10, and so on. This would be analogous to the mission milestone penalties.
II) Each level of duration imposes a -5 penalty, starting after the next level of duration. In this scenario, a single level of increase in duration above the highest achieved has no effect, with penalties beginning to accrue afterwards. A player who has achieved a B duration would be able to attempt a C duration mission with no penalty, or a D duration mission at a -5 penalty, and so on.

If (I) it would probably be best to preserve the Manned suborbital, Manned orbital, and Manned orbital EVA exemptions, lest early missions become too hard (sorry, Yuri).

As for the player difficulty, the game **does not** add a penalty to each mission on harder levels, at least as implemented. Instead, it reduces the total milestone penalty 2, 1, or 0 points for Level 1, Level 2, and Level 3 respectively. This can be verified by pulling up a new game and checking the Orbital Satellite penalty.

**Comment by peyre** [comment]
Ah, I've never played at higher than Level 1 (it's tough enough, thank you), so it may be failing to implement penalties that it should, without me knowing it.

The way the duration penalty system is supposed to work is that you get a -5 for every duration level you skip, and moving ahead one level should come with no penalty.  The game shouldn't penalize you for moving ahead incrementally, but it should ding you for charging ahead too fast.  So:

- There shouldn't be a duration penalty, ever, for flying your first Duration A mission.  Milestone penalties can apply (like if you skip the suborbital and go straight to orbital), but no duration penalties because you're not skipping any duration levels.
- If you've flown Duration A: no penalty for flying Dur B, but -5 for C, -10 for D, etc.
- If you've flown Duraiton B: no penalty for flying Dur C, but -5 for going straight to D.  Etc.

**Comment by rnyoakum** [comment]
I was reading over the original PDF manual, and was looking at the section on milestones:

> Durs. B and C must be completed in that order after manned orbitals and before a manned lunar pass; duration D must be completed before manned lunar orbital.
...
_Examples
If the first mission attempted is manned orbital, the penalty is -9% on all aspects of the mission. A  manned orbital 'B' has been successfully completed. If a manned lunar pass were attempted, it  would be -14% on all aspects of the mission._

For the Lunar Pass example, a lunar pass is a duration 'C' flight, and the Lunar Satellite Fly-By, Lunar Probe Landing, and Manned Lunar Pass objectives would be uncompleted, leaving a -5 duration penalty even with a completed 'B' duration. This suggests a duration penalty should remain until the duration C is achieved. The milestone text supports this:

> Primary Milestones are steps that must be completed in sequence. Durs. B and C must be completed in that order after manned orbitals and _before_ a manned lunar pass; _[Emphasis mine]_

For the example of the Manned Orbital, the penalty suggests that either Duration A is exempt from this milestone system, or (as the code seemed to suggest) the particular missions Manned suborbital, Manned orbital, and Manned orbital EVA are exempt.

My concern is that if duration penalties aren't applied until the duration is achieved, it eliminates the point of duration missions - that they are a simple way of achieving the duration milestone. A player gets Duration (A) from doing a manned flight, need a single duration (B), but then get C for free with a manned lunar pass and won't need anything else before a Lunar Landing. That cuts two duration missions from the flight schedule. It also reflects that the extra duration is something new, and new things are always riskier in spaceflight. Finally, it has the added benefit of treating Duration milestones similarly to Prestige milestones.

This is a big issue, and I want to make sure the original sources are considered and we're satisfied with game balance. Having said that, it's your call.

**Comment by peyre** [comment]
Hmm, strange.  I think his examples may be in error.  As he says on p.32, milestone penalties apply only to milestones that are _skipped_, and page 10 says that "penalties on a particular launch could include steps skipped in mission milestones and missed duration steps, so I'm sure that duration penalties are supposed to work the same way as milestone penalties.  If you've flown Dur B you should be able to schedule a Dur C without penalty - but if you went straight from B to D, you'd get a -5.

In his first example, your first three milestones are
- Orbital Satellite
- Suborbital / person in space (Dur A)
- Orbital (Dur A)
If your first mission was an Orbital Satellite you wouldn't be penalized, since you aren't skipping any steps.  If your first was a Suborbital, you'd be penalized -3 for skipping the satellite.  If your first mission was an Orbital, you should be penalized -6 for skipping the satellite and suborbital.  And no duration penalty since you're going from none to A.  I think he was mistaken about it being -9; he must have added all three steps instead of just two, since in this example you're only _skipping_ two steps.

After a Duration B, we have the following:
- Lunar Flyby
- Lunar Probe Landing
- Lunar Pass (Dur C)
In that example, you're skipping the Lunar Flyby and Probe Landing, but since you're going from Dur B to C you're not skipping a duration step, so the penalty should be -6, unless I'm missing something (which I have the feeling I might be doing).  I don't know where -14 came from, but I suspected that, again, he added all three milestone steps instead of just two, and added a duration penalty that (I think) shouldn't be there.

After a Duration D you'd have
- Lunar Flyby
- Lunar Probe Landing
- Lunar Pass
- Lunar Orbital
- Lunar Landing
which would be skipping four steps so the penalty should be -12%, so his third example is correct.

Also, the game adds a -1 penalty to a lot of missions, basically when you're doing something new.

**Comment by peyre** [comment]
_Waaait a minute, I **was** missing something._ 

Ok, I think I got milestone steps right, and the game does them like it should.  However, duration is a little different.  What I said was right, I think, about duration tests.  So if you've flown the Orbital and then schedule an Orbital Duration B there should be no penalty.

However, milestone missions should require that their duration level has been achieved.  So in his second example, you need to have accomplished Duration C before flying the Lunar Pass or you get the -5 on it.  So the penalty for the second example should be -11.

That brings us back to the first example.  The Suborbital (or Orbital) is a Dur A mission, and if you haven't flown a mission yet, you're at no duration level.  I think we probably should exempt Duration A from the duration penalty system; it doesn't make much sense (to me) that you face a heavy penalty on your first manned mission when by definition you can't have flown a Dur A prior to it.  The game would be penalizing you for something you _can't avoid_ skipping.

**Comment by peyre** [comment]
Incidentally, I was thinking along these lines re. allowing the player to downgrade the duration level of a duration mission.  On duration tests (and any mission that you've assigned a higher duration level to than it requires), an extra button might appear in Mission Control.  The reason I think this would be important (if it's practical to implement) is that duration tests sometimes fail.  Say you assemble a Dur C and schedule a Dur D for next turn, and the C mission has to cut the Dur step short and you're still at level B.  You're faced with the choice of flying your Dur D at a -5 penalty or scrubbing it.  It would be handy if you could just change it to a C mission.

![DurationDowngrade](https://user-images.githubusercontent.com/3118014/65567133-ed460700-df09-11e9-809e-c300fe6c72b2.png)

**Comment by rnyoakum** [comment]
I think you are right, the game handles milestones as it should (not considering the idea raised in #250). In my reading, when the game says "for each milestone skipped" it actually intends "for each milestone not yet achieved." To remark on some of your comments:

> Also, the game adds a -1 penalty to a lot of missions, basically when you're doing something new, so in the game itself the penalty is likely to show as -7 instead of -6, etc.

The -1 penalty is actually a -3 milestone penalty. It's just the game reduces the milestone penalty total by -2/-1 points when you're playing on Level 1/Level 2 difficulty. Hence, the -1 penalty on the first Orbital Satellite is actually the -3 Orbital Satellite milestone penalty, with a couple points refunded.

> However, milestone missions should require that their duration level has been achieved. So in his second example, you need to have accomplished Duration C before flying the Lunar Pass or you get the -5 on it. So the penalty for the second example should be -11.

In keeping with the other examples, it should be -14:
- Lunar Flyby Prestige (-3)
- Lunar Probe Landing Prestige (-3)
- Lunar Pass Prestige (-3)
- Duration C (-5)

> What I said was right, I think, about duration tests. So if you've flown the Orbital and then schedule an Orbital Duration B there should be no penalty.
> 
> However, milestone missions should require that their duration level has been achieved. So in his second example, you need to have accomplished Duration C before flying the Lunar Pass or you get the -5 on it.

Okay, this is interesting. My thinking was that the game should treat Duration milestones the same way as prestige milestones - it's something new, so there's a risk involved - with the exception of either

-  Duration A (because dogs/monkees). Thus any time you're attempting a new duration (except A) there would be a cumulative duration penalty (including for A), or
- Only missions 2,4, & 6 (Manned Suborbital, Manned Orbital, Orbital EVA); this being how the game originally handled it.

What it seems like you're proposing is that the game should allow you to perform the next duration step without penalty... unless you're trying for a milestone achievement. Am I understanding you correctly? That would be interesting.

**Comment by peyre** [comment]
Yes, that's what I'm proposing.  BARIS never applied duration penalties as long as you went from B to C to D etc., except that it always applied a duration penalty to your first duration mission, whatever it was.  That made your first duration mission a huge obstacle.  Then RIS at one point started applying a duration penalty to every duration level, and it made the game almost impossible to play.  If I'm not very much mistaken, there should be no penalty as long as you're running your duration tests one level at a time.  But if you try Lunar Pass (C) when you're still at level B, then there should be a penalty.

The middle example should still be -11 instead of -14, because you're only skipping two milestones.  If you went from Orbital to a Lunar Flyby there's no penalty because you're not skipping a step.  If you go from Orbital to Probe Landing, then you're skipping one step, the Flyby.  Going from the Orbital to Lunar Pass you're skipping two steps (plus duration levels skipped if you're not at C yet).

The -1% is for trying something new; one of the BARIS developers told me that.  I get the -3 milestone penalties even though I always play at level 1.  I can't find it now, but I'm pretty sure you get an extra -3 on each mission at level 2, and -6 at level 3.

**Comment by peyre** [comment]
This whole discussion gave me pause, so I wrote Michael McCarty to make sure how the system is supposed to work.  He confirmed that "milestone penalties were for the increased risk of trying to skip a step."

**Comment by rnyoakum** [comment]
Well, that clears most things up. With that in mind, I'll try to summarize how I think it works:

- A milestone penalty (-3) is applied for every milestone that has been **skipped**. Going for a new milestone does not, _by itself_, produce a _milestone_ penalty.
- A duration penalty (-5) is applied for every duration milestone that has been **skipped**. The new duration being attempted does not provoke a penalty, except
- Attempting a milestone mission without having completed the necessary duration for that mission **does** imply a duration penalty, because that duration is seen as a prerequisite (only relevant for Lunar Pass & beyond)
- Duration A will impose a penalty if **skipped**, but otherwise is not a prerequisite for any milestones because it is a milestone (Man In Space) itself, and so imposes no penalties on any Duration A milestone missions.

Please correct me if that's wrong. That just leaves the "new mission" penalty (-1 to -3):

1. What constitutes "trying something new?" Is it determined by a new milestone, unachieved prestige award, or mission code?
2. Is the mission considered "safe" after being attempted or being successfully completed?

**Comment by peyre** [comment]
Yes, you have that just right, how the system's supposed to work.

About the new mission penalty, I'm not sure exactly what counts it as a new mission, but I could send Michael a quick note and ask if you like.  I'm pretty sure it will have the -1 to -3 until it's successfully completed; failed attempts don't satisfy the requirement.  I think.

**Comment by peyre** [comment]
I asked Michael, and he wrote back:

> Once the mission is successfully flown then the penalty goes away.   I can see that even on a failed mission with that penalty, then something was learned making that next attempt slightly less risky.
> 
> I don't recall what a new mission would be classified as.  This would probably follow prestige.    You should choose to interpret it as you like to continue to improve the modeling of the risk.

So I guess maybe the penalty should only apply to milestone missions.  That would be in keeping with RIS's habit of hitting your prestige only if the mission could have provided additional prestige (that is, dummy missions don't affect prestige).  Whether we want the penalty to go away after a failed attempt, I don't have strong feelings about.  I'd be ok with it either way.

**Comment by rnyoakum** [comment]
I was coding the mission penalty as I described in my previous post, but had a concern:

- A Manned Earth Orbital would start with a -7 penalty. This comes from two skipped milestones (Orbital Satellite, Man In Space) and the new mission penalty. There's no duration penalty, because it's Duration A.
- A Manned Orbital Duration (B) would start with a -17 penalty. This comes from two skipped milestones, a skipped duration (A), _plus_ it would be applied the duration (B) penalty because it accomplishes a new milestone goal (Man in Space, Manned Orbital) and so is considered a milestone mission with a duration requirement.

Basically, any time a player wanted to do a manned mission with Duration B or more as their first manned mission it would apply all the penalties. Correct me if I'm wrong, but this seems exactly the situation you wanted to avoid. Instead, I was going to adjust the penalty from I described above to be:

- A duration penalty (-5) is applied for every duration milestone that has been **skipped**. The new duration being attempted does not provoke a penalty, except
- Manned Lunar Pass requires Duration C, and
- Manned Lunar Orbit & Manned Lunar Landing require Duration D

That tracks closer to the manual than what I had been discussing.

**Comment by peyre** [comment]
Yes, that's right, if you schedule a Dur B as your first mission (or even just your first manned mission), you would get one duration penalty (-5) because you're skipping Dur A.  You wouldn't get two duration penalties unless you made your first mission a Dur C.

> Basically, any time a player wanted to do a manned mission with Duration B or more as their first manned mission it would apply all the penalties. Correct me if I'm wrong, but this seems exactly the situation you wanted to avoid.
That's exactly right.  That's what BARIS used to do, and it used to drive me nuts because it made your first duration mission a big hurdle, and it wasn't supposed to do that anyway.

What you're outlining sounds right, yes.  Basically, any prestige first (eg Lunar Pass) whose minimum duration level (C in this case) will give you a duration penalty for every Dur level you haven't already reached, so if you've flown only Dur B you'll get a -5, but if you've only flown a Dur A, you'll get a -10 on the Lunar Pass.  But a plain Duration mission won't penalize you unless you're skipping a duration level.

**Comment by rnyoakum** [comment]
I've implemented the mission penalty system per the most recent comments. Let me know if there's anything you see that's unexpected, but I think this is handled.

Fixed per commit e51fbf66eb64d5711b430d7b9c45a1f563cfd37c.

**Comment by rnyoakum** [comment]
Hmm - I was looking over the code, and there might be an additional problem that's a part of this that you did (mostly) disable. The problem you described was that the duration penalty suddenly changed by 10 points instead of 5 when increasing the duration. This issue is handled. But there was also another issue, the one you mostly disabled, that would apply the duration penalty to the mission twice. If you'll follow:

When the Future Missions is trying to calculate the penalty on a mission, it calls a function in prest.cpp, PrestMin(). This calculates the mission penalty and returns it. This function contained the penalty logic that was causing the jump in visible duration penalty, and I corrected it.

When running the mission, the Launch function in mc.cpp runs:

```
MisDur(plr, Data->P[plr].Mission[mis].Duration);

if (MANNED[0] > 0 || MANNED[1] > 0 || mcode == 1 || mcode == 7 || mcode == 8) {
    MisSkip(plr, Find_MaxGoal());
}

MisRush(plr, Data->P[plr].Mission[mis].Rushing)
```

There are three penalties being applied to each mission in this code:

1. MisDur() calculates a mission duration penalty and applies it to every step of the mission.
2. MisSkip() calculates the mission penalty - using the PrestMin() function - and applies it to every step of the mission.
3. MisRush() calculates a rushing penalty, and applies it to every step of the mission.

MisDur() adds a duration penalty to every step, but PrestMin() _also_ calculates the duration penalty and via MisSkip is applying that penalty to every step. At some point you edited MisDur() so it wouldn't apply any penalty unless the penalty was at least three steps; I imagine that doesn't come up often, so it's practically as if it was completely disabled. I think that this is a remnant of old code, where the duration and prestige penalties were calculated separately. I tried to trace what was going on by looking through the code history, but ran into a hard wall at the initial check-in.

I'm pushing a commit to replace the mc.cpp code with

```
MisSkip(plr, Find_MaxGoal());
MisRush(plr, Data->P[plr].Mission[mis].Rushing);
```

Since the PrestMin() function used by MisSkip calculates the overall Duration + Prestige penalty for each mission, MisDur() can be eliminated. The previous check - any manned mission and Orbital Satellite, Lunar Flyby, and Lunar Probe - doesn't seem relevant, and the penalty for any mission should be whatever Future Missions is displaying (i.e. PrestMin) plus any rushing penalties.

**Comment by peyre** [comment]
Yes, what I did (making it not apply a penalty unless you were missing three steps) was a dirty hack to effectively disable the duration penalty system since I couldn't get it to work right.  Fixing it so it works right would be easily the better option.

I don't follow everything you mention up there; it has been some time since I messed with the duration penalty system.  But so much the better if it's fixed.

Ok, I've redownloaded and compiled the code, and it looks like you've essentially put it back the way I had it.  In this screenshot, I'm at Duration A, and I'm not being docked for planning a Duration C mission.
![NoPenalty](https://user-images.githubusercontent.com/3118014/66259300-3006ab00-e764-11e9-935e-8600eabaf208.png)

**Comment by rnyoakum** [comment]
That... is not what I was expecting. This is definitely a bug - I haven't gotten a strong test setup working yet, and didn't play far enough into the game to see this. The duration penalties are calculated correctly when starting a new game, but this suggests it stops working once Duration A is completed. I'll take another look at it.

**Comment by peyre** [comment]
Thanks!  I noticed that it is applying milestone penalties correctly, as far as I can tell.

**Comment by rnyoakum** [comment]
I believe I fixed the particular bug that kept durations > B from being reflected in the FM mission penalty (the actual mission used the correct penalty). As of commit d55bd23d9ddffde447eac74d74802dd98d80cd7d, the duration system should be functioning as detailed in this issue.
